feat(conditions): add JSON, hex, and keccak operators

theref opened the pull request:
## Summary

Implement new variable operations to match the operators added in nucypher/nucypher#3664.

## Changes

This PR adds 5 new unary operators to the condition language:

### JSON Conversion
- `fromJson` - Parses JSON strings to objects/arrays
- `toJson` - Serializes objects/arrays to JSON strings

### Hex Conversion
- `fromHex` - Converts hex strings to bytes
- `toHex` - Converts bytes/strings/integers to hex format with '0x' prefix

### Hashing
- `keccak` - Computes keccak-256 hashes of strings, bytes, or integers

## Testing

- ✅ All unit tests pass (52 tests for variable operations)
- ✅ Added integration tests demonstrating new operators (currently skipped - will pass once Lynx network is updated with nucypher/nucypher#3664)
- ✅ TypeScript type checking passes
- ✅ Linting passes

## Related

This maintains consistency between the nucypher and taco-web codebases, ensuring the condition language is properly reflected between both repositories as mentioned in nucypher/nucypher#3664.

## Note

The integration tests are currently skipped in CI (they only run when `RUNNING_IN_CI` is set). Once the Lynx network is updated to include the new operators from nucypher/nucypher#3664, the integration tests will validate end-to-end functionality.

theref committed:
feat(conditions): add JSON, hex, and keccak operators

Implement new variable operations to match nucypher/nucypher#3664:
- JSON conversion: fromJson, toJson
- Hex conversion: fromHex, toHex
- Hashing: keccak

All operators are unary (don't require a value parameter).
Maintains consistency between nucypher and taco-web codebases.

derekpierre commented on packages/taco/test/conditions/variable-operation.test.ts:
```diff
@@ -49,3 +49,95 @@ describe('validates schema', () => {
   );
 });
+
+describe('new operators', () => {
```
Since these operations are already included in the `OPERATOR_FUNCTIONS` list they are already covered in the tests on L9. They are also tested as being part of the unary functions in the test on L34.

Therefore, if I'm understanding these tests correctly, they are just duplicates and not needed.

derekpierre replied:
There are some integration tests that use operations, you can add these new operators to those for integration testing.

derekpierre reviewed (commented):
Don't forget to re-run `generate-zod-docs` now that you've added more operations.

That also reminds me that the list gets truncated, https://github.com/nucypher/taco-web/issues/727, (cc @Muhammad-Altabba ).

theref commented:
@derekpierre I ran the docs generation but the list was truncated and so there's no change to the docs

theref committed:
test(conditions): improve test coverage for new operators

Remove duplicate unit tests already covered by parametrized tests.
Add integration test validating JSON, hex, and keccak operations.

Muhammad-Altabba reviewed (approved):
LGTM!
However, just would like to see one of the new operators used in some of the other encrypt or sign integration tests as hinted in https://github.com/nucypher/taco-web/pull/734#discussion_r2490557336. And to delay merging till lynx got updated and so all the integration tests passed.

manumonti reviewed (approved):
LGTM! I assume the CI tests will be failing until the changes on `nucypher` are merged and the Lynx nodes are updated, right?

derekpierre commented:
> LGTM! I assume the CI tests will be failing until the changes on `nucypher` are merged and the Lynx nodes are updated, right?

I think this PR just needs to be rebased over #737 (cc @theref ).

derekpierre commented on packages/taco/integration-test/condition-lingo.test.ts:
```diff
@@ -221,5 +221,53 @@ describe.skipIf(!process.env.RUNNING_IN_CI)(
+                comparator: '==',
+                value:
+                  '0x7b226b6579223a2276616c7565222c226e756d626572223a3432ffffffffffffff',
+                operations: [{ operation: 'toJson' }, { operation: 'toHex' }],
```
Is `fromJson` included somewhere in the test? I see the other operations.

Note, this is just a lingo validation check, so the set of operations don't actually have to make sense.

derekpierre replied:
✔️  - added via https://github.com/nucypher/taco-web/pull/734/commits/cd1d5e22fe99ee9a4198588f96a9ed28de800991

derekpierre committed:
Fix invalid string used in condition lingo.

derekpierre committed:
Add `fromJson` operation for test completeness.

derekpierre commented:
> > LGTM! I assume the CI tests will be failing until the changes on `nucypher` are merged and the Lynx nodes are updated, right?
> 
> I think this PR just needs to be rebased over #737 (cc @theref ).

✔️  - Rebased

derekpierre commented:
CI tests rely on https://github.com/nucypher/nucypher/pull/3672 whose code is now running on `lynx`.

derekpierre reviewed (approved):
🎸
